tests/components/homekit/test_type_humidifiers.py

```diff
@@ -0,0 +1,419 @@
+
+    char_active_iid = acc.char_active.to_HAP()[HAP_REPR_IID]
+
+    hk_driver.set_characteristics(
+        {
+            HAP_REPR_CHARS: [
+                {
+                    HAP_REPR_AID: acc.aid,
+                    HAP_REPR_IID: char_active_iid,
+                    HAP_REPR_VALUE: 1,
+                },
+            ]
+        },
+        "mock_addr",
+    )
+
+    await hass.async_block_till_done()
+    assert len(call_turn_on) == 1
+    assert call_turn_on[0].data[ATTR_ENTITY_ID] == entity_id
+    assert acc.char_active.value == 1
+    assert len(events) == 1
+    assert events[-1].data[ATTR_VALUE] == "Active to 1"
+
+    call_turn_off = async_mock_service(hass, DOMAIN, SERVICE_TURN_OFF)
+
+    hk_driver.set_characteristics(
+        {
+            HAP_REPR_CHARS: [
+                {
+                    HAP_REPR_AID: acc.aid,
+                    HAP_REPR_IID: char_active_iid,
+                    HAP_REPR_VALUE: 0,
+                },
+            ]
+        },
+        "mock_addr",
+    )
+
+    await hass.async_block_till_done()
+    assert len(call_turn_off) == 1
+    assert call_turn_off[0].data[ATTR_ENTITY_ID] == entity_id
+    assert acc.char_active.value == 0
+    assert len(events) == 2
+    assert events[-1].data[ATTR_VALUE] == "Active to 0"
+
+
+async def test_hygrostat_get_humidity_range(hass, hk_driver):
+    """Test if humidity range is evaluated correctly."""
+    entity_id = "humidifier.test"
+
+    hass.states.async_set(
+        entity_id, STATE_OFF, {ATTR_MIN_HUMIDITY: 40, ATTR_MAX_HUMIDITY: 45}
+    )
+    await hass.async_block_till_done()
+    acc = HumidifierDehumidifier(
+        hass, hk_driver, "HumidifierDehumidifier", entity_id, 1, None
+    )
+    hk_driver.add_accessory(acc)
+
+    await acc.run_handler()
+    await hass.async_block_till_done()
+
+    assert acc.char_target_humidity.properties[PROP_MAX_VALUE] == 45
+    assert acc.char_target_humidity.properties[PROP_MIN_VALUE] == 40
+
+
+async def test_humidifier_with_linked_humidity_sensor(hass, hk_driver):
+    """Test a humidifier with a linked humidity sensor can update."""
+    humidity_sensor_entity_id = "sensor.bedroom_humidity"
+
+    hass.states.async_set(
+        humidity_sensor_entity_id,
+        "42.0",
+        {
+            ATTR_DEVICE_CLASS: DEVICE_CLASS_HUMIDITY,
+            ATTR_UNIT_OF_MEASUREMENT: UNIT_PERCENTAGE,
+        },
+    )
+    await hass.async_block_till_done()
+    entity_id = "humidifier.test"
+
+    hass.states.async_set(entity_id, STATE_OFF)
+    await hass.async_block_till_done()
+    acc = HumidifierDehumidifier(
+        hass,
+        hk_driver,
+        "HumidifierDehumidifier",
+        entity_id,
+        1,
+        {CONF_LINKED_HUMIDITY_SENSOR: humidity_sensor_entity_id},
+    )
+    hk_driver.add_accessory(acc)
+
+    await acc.run_handler()
+    await hass.async_block_till_done()
+
+    assert acc.char_current_humidity.value == 42.0
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5795327)